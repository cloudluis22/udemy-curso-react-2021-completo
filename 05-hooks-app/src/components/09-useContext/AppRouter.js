import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from 'react-router-dom'

import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>

            <NavBar />

            <div>

                <Switch>

                    <Route exact path="/" component={HomeScreen} />
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />

                    {/* Si no encuentra una ruta espec√≠fica, redirige a la ruta por defecto. */}
                    <Redirect to='/' />

                </Switch>

            </div>

        </Router>
    )
}
