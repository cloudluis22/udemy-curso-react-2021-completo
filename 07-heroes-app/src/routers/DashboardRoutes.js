import React from 'react'
import {

  BrowserRouter as Router,
  Switch,
  Route,
  Redirect

} from "react-router-dom";
import { DCScreen } from '../components/dc/DCScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/NavBar';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container mt-2">
                <Switch>

                    <Route exact path="/marvel" component={MarvelScreen} />
                    <Route exact path="/hero/:heroId" component={HeroScreen} />
                    <Route exact path="/dc" component={DCScreen} />
                    <Route exact path="/search" component={SearchScreen} />

                    <Redirect to="/marvel" />
                </Switch>
            </div>


        </>
    )
}
