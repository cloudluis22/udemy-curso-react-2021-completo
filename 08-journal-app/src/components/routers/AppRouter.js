import React, { useEffect, useState } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { firebase } from '../../firebase/firebaseConfig';
import { useDispatch } from 'react-redux';

import AuthRouter from './AuthRouter';
import JournalScreen from '../journal/JournalScreen';
import { login } from '../../actions/auth';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { startLoadingNotes } from '../../actions/notes'

const AppRouter = () => {

    const dispatch = useDispatch();
    const [checking, setChecking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false);

    console.log(isLoggedIn);

    useEffect( () => {

        firebase.auth().onAuthStateChanged( async (user) => {

            if (user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);

                dispatch( startLoadingNotes(user.uid) );
            }
            else {
                setIsLoggedIn(false);
            }

            // Una vez se autentica ya no se está revisando si el usuario está siendo autenticado.
            setChecking(false)
        })

    }, [dispatch, setChecking, setIsLoggedIn]);

    if (checking) {
        return (
            <div className="spinner round"></div>
        )
    }

    return (
        <Router>
            <div>
                <Switch>

                    <PublicRoute isLoggedIn={ isLoggedIn } component={ AuthRouter } path="/auth" />
                    <PrivateRoute exact isLoggedIn={ isLoggedIn } path="/" component={ JournalScreen } />

                    <Redirect to="/auth/login" />

                </Switch>
            </div>
        </Router>
    )
}

export default AppRouter;