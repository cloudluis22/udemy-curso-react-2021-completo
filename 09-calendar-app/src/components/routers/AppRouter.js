import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  BrowserRouter as Router,
  Switch,
  Redirect
} from "react-router-dom";
import { startChecking } from '../../actions/auth';
import { LoginScreen } from '../auth/LoginScreen';
import { CalendarScreen } from '../calendar/CalendarScreen';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const dispatch = useDispatch();
    const { checking, uid } = useSelector(state => state.auth);

    useEffect(() => {
        dispatch( startChecking() );
    }, [dispatch]);

    if (checking) {
        return <h5> Espere... </h5>
    }

    return (
        <Router>
            <div>
                <Switch>

                    <PublicRoute exact path="/login" isLoggedIn={ !!uid } component={ LoginScreen } />
                    <PrivateRoute exact path="/" isLoggedIn={ !!uid  } component={ CalendarScreen } />

                    <Redirect to="/" />

                </Switch>
            </div>
        </Router>
    )
}
