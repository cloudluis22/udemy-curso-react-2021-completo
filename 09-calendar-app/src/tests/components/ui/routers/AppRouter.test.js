import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme from 'enzyme';

import '@testing-library/jest-dom';
import { AppRouter } from '../../../../components/routers/AppRouter';


Enzyme.configure({ adapter: new Adapter() });

const middlewares = [ thunk ];
const mockStore = configureStore(middlewares);

describe('Pruebas en el componente de "<AppRouter />."', () => {

    test('El componente debe renderizarse correctamente y generar la snapshot. ', () => {

        const initState = {
            auth: {
                checking: true
            }
        };

        const store = mockStore(initState);

        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar la ruta pública si no está autenticado.', () => {

        const initState = {
            auth: {
                checking: false,
                uid: null
            }
        };

        const store = mockStore(initState)

        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect(wrapper.find('.login-container').exists()).toBe(true);

    });

    test('Debe de mostrar la ruta priva si está autenticado.', () => {

        const initState = {
            ui: {
                modalOpen: false
            },
            calendar: {
                events: []
            },
            auth: {
                checking: false,
                uid: '123',
                name: 'Test Man'
            }
        };

        const store = mockStore(initState)

        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect(wrapper.find('.calendar-screen').exists()).toBe(true);

    });

});