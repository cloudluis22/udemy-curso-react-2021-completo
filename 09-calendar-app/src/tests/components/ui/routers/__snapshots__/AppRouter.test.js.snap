// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de "<AppRouter />." El componente debe renderizarse correctamente y generar la snapshot.  1`] = `ReactWrapper {}`;
